src/Features/VisualBasic/Portable/ConvertIfToSwitch/VisualBasicConvertIfToSwitchCodeRefactoringProvider.vb

```diff
@@ -1,197 +1,154 @@
-                Dim elseBlockStatements As SyntaxList(Of StatementSyntax)?
-
-                Dim singleLineIf = TryCast(node, SingleLineIfStatementSyntax)
-                If singleLineIf IsNot Nothing Then
-                    Yield (singleLineIf.Condition, singleLineIf.Statements)
-                    elseBlockStatements = singleLineIf.ElseClause?.Statements
-                Else
-                    Dim multiLineIf = DirectCast(node, MultiLineIfBlockSyntax)
-                    Yield (multiLineIf.IfStatement.Condition, multiLineIf.Statements)
-                    For Each elseIfBlock In multiLineIf.ElseIfBlocks
-                        Yield (elseIfBlock.ElseIfStatement.Condition, elseIfBlock.Statements)
-                    Next
-
-                    elseBlockStatements = multiLineIf.ElseBlock?.Statements
-                End If
-
-                If elseBlockStatements.HasValue Then
-                    Yield (Nothing, elseBlockStatements.Value)
-                End If
-            End Function
+            Public Overrides ReadOnly Property SupportsCaseGuard As Boolean
+                Get
+                    Return False
+                End Get
+            End Property
 
-            Private Function AnalyzeControlFlow(statements As SyntaxList(Of StatementSyntax)) As ControlFlowAnalysis
-                Return If(statements.IsEmpty, Nothing, _semanticModel.AnalyzeControlFlow(statements.First(), statements.Last()))
-            End Function
+            Public Overrides ReadOnly Property SupportsRangePattern As Boolean
+                Get
+                    Return True
+                End Get
+            End Property
 
-            Protected Overrides Function EndPointIsReachable(ifStatement As ExecutableStatementSyntax) As Boolean
-                Return If(AnalyzeControlFlow(ifStatement.GetStatements())?.EndPointIsReachable, False)
-            End Function
+            Public Overrides ReadOnly Property SupportsTypePattern As Boolean
+                Get
+                    Return False
+                End Get
+            End Property
 
-            Protected Overrides Function UnwrapCast(expression As ExpressionSyntax) As ExpressionSyntax
-                Return expression
-            End Function
+            Public Overrides ReadOnly Property SupportsSourcePattern As Boolean
+                Get
+                    Return False
+                End Get
+            End Property
 
-            Protected Overrides Function CreateSwitchStatement(ifStatement As ExecutableStatementSyntax, expression As ExpressionSyntax, sectionList As List(Of SyntaxNode)) As SyntaxNode
-                Return VisualBasicSyntaxGenerator.Instance.SwitchStatement(expression, sectionList)
-            End Function
+            Public Overrides ReadOnly Property SupportsRelationalPattern As Boolean
+                Get
+                    Return True
+                End Get
+            End Property
```

Comment:
so i can't see where these properties are even used.  are they necessary any more?

Reply:
I think you haven't reviewed the Analyzer class.